test/relations.test.js

```diff
@@ -31,6 +31,60 @@ describe('loopback:relation generator', function() {
       });
   });
 
+   it('check naming conflict for relation', function(done) {
+    var propertyGenerator = givenPropertyGenerator();
+    var relationGenerator = givenRelationGenerator();
+
+    // create a Car model and give it a property called 'year'
+    helpers.mockPrompt(propertyGenerator, {
+      model: 'Car',
+      name: 'year',
+      customType: '', // temporary workaround for
+                      // https://github.com/yeoman/generator/issues/600
+      customItemType: '', // temporary workaround for
+                          // https://github.com/yeoman/generator/issues/600
+      type: 'boolean',
+      required: 'true'
+    });
+
+    propertyGenerator.run(function() {
+      var definition = readJsonSync('common/models/car.json');
+      var props = definition.properties || {};
+      expect(props).to.have.property('year');
+      expect(props.year).to.eql({
+        type: 'boolean',
+        required: true
+      });
+    });
+
+    // name a relationship as 'year', 
+    // should get error here but the validate function doesn't work
+    // even I name it as 'year@@@', it passes the test
+    // validateName doesn'n work either
+    // but they all work fine when I run slc loopback:relation in cmd
```

Comment:
Apparently the mock prompt does not run validations - see [source code](https://github.com/yeoman/generator/blob/a7d8dd0fcaed7a12e440579ecafaf1287d326f1f/lib/test/adapter.js#L12-L41).

I am proposing to extract the validate method into helpers (e.g. `helpers.validateRelationName`) and test it in isolation, like we test `validateName` - see [test/helpers.js](https://github.com/strongloop/generator-loopback/blob/a884c0b27314e3b2602fff1c3b6f75b602e0b57a/test/helpers.test.js).